work26/js/contents.js

```diff
@@ -0,0 +1,305 @@
+sportsContents.id = "js-sports"
+japanContents.id = "js-japan"
+
+for (let i = 0; i < contents.length; i++) {
+  contentsWrap.appendChild(contents[i])
+  contents[i].classList.add('content')
+  contentsInners[i].classList.add('content__inner')
+  titleWraps[i].classList.add('titleWrap')
+  body.appendChild(contentsWrap)
+}
+
+async function fetchArticle(url) {
+  try {
+    const response = await fetch(url)
+    const json = await response.json()
+    return json.articles
+  } catch {
+    tabs.textContent = 'ただいまサーバー側がぶっこわれています。'
+  } finally {
+    console.log('fetchArticle run')
+  }
+}
+
+async function init(url) {
+  let articles
+  try {
+    articles = await fetchArticle(url)
+  } catch (e) {
+    console.log(e);
+  } finally {
+    console.log('init executed');
+  }
+
+  createTabs(articles)
+  createTitles(articles)
+  createImages(articles)
+  checkContentsIsInit(articles)
+}
+init(fetchContentsURL)
+
+function createTabs(articles) {
+  const tabFrag = document.createDocumentFragment()
+  const tabList = new Set()
+
+  for (const article of articles) {
+    const { category, isInit } = article
+    const tabItem = document.createElement('li')
+    tabItem.textContent = category
+    tabItem.classList.add('tab__item')
+    tabFrag.appendChild(tabItem)
+    tabs.appendChild(tabFrag)
+    tabList.add(tabItem)
+
+    if (checkTabIsActive(isInit)) {
+      addActiveClassName(tabItem)
+    }
+
+    addTabId(category, tabItem)
+    tabClickAction(tabList)
+  }
+}
+
+function checkTabIsActive(isInit) {
+  if (isInit) return true
+}
+
+function addTabId(category, tabItem) {
+  if (category === 'ニュース') {
+    tabItem.dataset.id = 'js-news'
+  } else if (category === '経済') {
+    tabItem.dataset.id = 'js-economy'
+  } else if (category === 'エンタメ') {
+    tabItem.dataset.id = 'js-entertainment'
+  } else if (category === 'スポーツ') {
+    tabItem.dataset.id = 'js-sports'
+  } else if (category === '国内') {
+    tabItem.dataset.id = 'js-japan'
+  }
+}
+
+function tabClickAction(tabList) {
+  tabList.forEach(clickedTab => {
+    clickedTab.addEventListener('click', () => {
+      tabList.forEach(tab => {
+        tab.classList.remove('active')
+      })
+
+      clickedTab.classList.add('active')
+
+      contents.forEach(content => {
+        content.classList.remove('active')
+      })
+
+      document.getElementById(clickedTab.dataset.id).classList.add('active')
+    })
+  })
+}
+
+function checkContentsIsInit(articles) {
+  for (const article of articles) {
+    const { category, isInit } = article
+
+    if (category === 'ニュース' && isInit) {
+      addActiveClassName(newsContents)
+    } else if (category === '経済' && isInit) {
+      addActiveClassName(economyContents)
+    } else if (category === 'エンタメ' && isInit) {
+      addActiveClassName(entertainmentContents)
+    } else if (category === 'スポーツ' && isInit) {
+      addActiveClassName(sportsContents)
+    } else if (category === '国内' && isInit) {
+      addActiveClassName(japanContents)
+    }
```

Comment:
isIntじゃなかったら早期リターンできますか？

```suggestion
　if(isInit) return
    if (category === 'ニュース' ) {
      addActiveClassName(newsContents)
    } else if (category === '経済' ) {
      addActiveClassName(economyContents)
    } else if (category === 'エンタメ') {
      addActiveClassName(entertainmentContents)
    } else if (category === 'スポーツ') {
      addActiveClassName(sportsContents)
    } else if (category === '国内') {
      addActiveClassName(japanContents)
    }
```

あとはswitch文か、先程のIMOで
これはこのelse ifに引っかからない場合はどんな処理をすべきですか

Reply:
早期リターンできました。
if文をswitch文に置き換え、default句で
```
tabs.textContent = 'ただいまサーバー側がぶっこわれています。'
```
としています。ただ、defaultで壊れていると表記するのも違和感があるのですが...
先生の意図に即した内容でしょうか？

e15b9e2